it/controllers/benefits/TelephoneBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,510 @@
+        " data when the user chooses no" which {
+
+        lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.no)
+
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          insertCyaData(employmentUserData(isPrior = true, cyaModel("employerName", hmrc = true,
+            Some(benefits(fullUtilitiesAndServicesModel)))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(telephoneQuestionPageUrl(taxYearEOY), body = form, follow = false, welsh = user.isWelsh,
+            headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "redirects to the check your details page" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe
+            Some(s"/income-through-software/return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=$employmentId")
+          lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+          cyamodel.employment.employmentBenefits.flatMap(_.utilitiesAndServicesModel.flatMap(_.utilitiesAndServicesQuestion)) shouldBe Some(true)
+          cyamodel.employment.employmentBenefits.flatMap(_.utilitiesAndServicesModel.flatMap(_.telephoneQuestion)) shouldBe Some(false)
+          cyamodel.employment.employmentBenefits.flatMap(_.utilitiesAndServicesModel.flatMap(_.telephone)) shouldBe None
+        }
+
+      }
+
+      "redirect to check your benefits and update the telephoneQuestion to yes and when the user chooses yes" which {
+
+        lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.yes)
+
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          insertCyaData(employmentUserData(isPrior = true, cyaModel("employerName", hmrc = true,
+            Some(benefits(fullUtilitiesAndServicesModel.copy(telephoneQuestion = Some(false)))))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(telephoneQuestionPageUrl(taxYearEOY), body = form, follow = false, welsh = user.isWelsh,
+            headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "redirects to the check your details page" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe
+            Some(s"/income-through-software/return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=$employmentId")
+          lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+          cyamodel.employment.employmentBenefits.flatMap(_.utilitiesAndServicesModel.flatMap(_.utilitiesAndServicesQuestion)) shouldBe Some(true)
+          cyamodel.employment.employmentBenefits.flatMap(_.utilitiesAndServicesModel.flatMap(_.telephoneQuestion)) shouldBe Some(true)
+          cyamodel.employment.employmentBenefits.flatMap(_.utilitiesAndServicesModel.flatMap(_.telephone)) shouldBe Some(100)
```

Comment:
I know the test passes, but isn't it a case that should never happen to have amount after the value of the question has been changed from 'no' to 'yes'? I would probably remove line 424.

Reply:
I think this is a valid test. The controller wipes out this value if the user selects  no
This tests confirms the value is not wiped out if the user selects yes (e.g. if the controller changes to wipe it out on both this test would pick it up)

It doesn't have to be going from no to yes. It could be just going from yes to yes too

Reply:
Yes, good point. Thank you